### ISSUE TITLE
MongoDB extension support

### ISSUE THREAD

**Issue by ShockedPlot7560** [pmmp/php-build-scripts#145]
This binary contains direct support for mysql and sqlite however as soon as we want to use another database such as mongo, it is mandatory to create our own fork and to modify it ourselves.

Seeing the increasing demand for this extension, I think it would be a good idea to add it. In any case, it is always possible to disable it in the php.ini

**Comment by AkmalFairuz** [comment]
I don't recommend to use MongoDB in PM server.

The extension was designed for web application, the MongoDB connection will close after the script end. There is no way to close the Mongo connection like `mysqli_close($conn)` (this will cause memory leak) unless you're using the old Mongo extension.

https://github.com/mongodb/mongo-php-library/issues/600

**Comment by SOF3** [comment]
would it be better to access mongodb with a userspace library instead?

**Comment by dktapps** [comment]
> I don't recommend to use MongoDB in PM server.
> 
> The extension was designed for web application, the MongoDB connection will close after the script end. There is no way to close the Mongo connection like `mysqli_close($conn)` (this will cause memory leak) unless you're using the old Mongo extension.
> 
> [mongodb/mongo-php-library#600](https://github.com/mongodb/mongo-php-library/issues/600)

I don't think this is a concern. The reason they implemented it like this is so that different PHP web requests can reuse the same client without having to reestablish connections. In the PM context, every thread is a request.

Unless you have a habit of establishing large numbers of MongoDB connections to many different databases, I don't really think this is a problem.

**Comment by SOF3** [comment]
so is the mongodb instance freed when a thread exits? if that is the case, it sounds reasonable.

**Comment by dktapps** [comment]
So far as I understand, no. But in any case, I don't think it matters.

**Comment by SOF3** [comment]
For example, in terms of network errors, would it still leak a resource even though connection breaks?

**Comment by ShockedPlot7560** [comment]
In any case, it is up to the developer in this case to pay attention to all the tools he uses and to get informed about them.  
Simply, at the moment, the only way to use a db mongo is either to find a wobbly fork with the extension, or to change DB.

**Comment by SOF3** [comment]
Note that the use of mongodb itself is highly controversial due to its lack of consistency in its opinionated preference over simplicity.
